db/migrate/20210114105242_create_gifts.rb

```diff
@@ -1,15 +1,12 @@
 class CreateGifts < ActiveRecord::Migration[6.1]
   def change
     create_table :gifts do |t|
-      t.string :title, null: false
-      t.text :description
-      t.belongs_to :recipient, null: false
-      t.belongs_to :added_by, null: true
+      t.string     :title,      null: false
+      t.text       :description
+      t.belongs_to :recipient,  null: false, foreign_key: { to_table: 'users' }
+      t.belongs_to :added_by,   foreign_key: { to_table: 'users' }
```

Comment:
Oh, I understood why you added foreign keys separately, Rails cannot understand to which table create a constraint, so it was ok, sorry. Well, now it's even better :)